### ISSUE TITLE
[a11y] Form field examples lack a form wrapper

### ISSUE THREAD

**Issue by dpersing** [Shopify/polaris#1113]
# Issue summary

When we demonstrate using form fields in examples, they are not consistently wrapped in `<form>` elements. This can cause some inconsistent behaviors with screen readers that have different interaction/focus and reading modes, like NVDA.

## Expected behavior

Form fields can always be activated with the keyboard.

## Actual behavior

Checking and unchecking checkboxes in the checkbox and choice list components doesn't work unless NVDA (with Firefox) is put manually into forms/interaction mode. If the checkbox is wrapped in a `<form>`, the functionality is normal.

[Learn more about keyboard shortcuts for NVDA.](https://dequeuniversity.com/screenreaders/nvda-keyboard-shortcuts)

## Steps to reproduce the problem

1. Navigate to any [checkbox](https://polaris.shopify.com/components/forms/checkbox) or [choice list](https://polaris.shopify.com/components/forms/choice-list) that uses checkboxes in examples, using Firefox on Windows.
1. Using NVDA, tab to the checkboxes. When a checkbox has focus, press the `space` key to check it.
1. Press `insert` + `space` on Windows or number pad `0` + `space` on a Mac keyboard to turn focus mode on manually. 
1. Attempt to check the box again.

## Specifications

- Are you using the React components? (Y/N): Yes
- Polaris version number: 3.9.0
- Browser: Firefox with NVDA
- Device: MacOS (VM)
- Operating System: Windows 10

**Comment by AndrewMusgrave** [comment]
Is this something the styleguide should handle? Similar to how it wraps the examples in `AppProvider`. This way we can simplify our examples.

**Comment by dpersing** [comment]
I think that would be great, @AndrewMusgrave!
